src/NuGet.Core/NuGet.Packaging/Rules/XdtTransformsInPackageReferenceProjectRule.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Rules
+{
+    internal class XdtTransformInPackageReferenceProjectRule : IPackageRule
+    {
+        private const string ConfigTransformExtension = ".transform";
```

Comment:
In some cases, we also have `.install.xdt` and `.uninstall.xdt` I'm not sure if we want to warn about those as well?